### PR TITLE
Introduce CSS custom properties to `_TopUnreadMessagesBar.pcss` 

### PR DESCRIPTION
**Pull request by luixxiul**
Closes https://github.com/vector-im/element-web/issues/25282

This PR suggests to update `_TopUnreadMessagesBar.pcss` by introducing CSS custom properties, splitting structural definitions from their values. This way it will become possible to access and manipulate those values via TypeScript to change, for example, the border color and size, maintaining the style structure.

type: task

| |Image|
|-|---|
|TopUnreadMessageBar||
|`mx_TopUnreadMessagesBar_scrollUp`<br />cascading||
|`mx_TopUnreadMessagesBar_markAsRead`<br />cascading||

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by luixxiul** [comment]
Trying to test https://github.com/matrix-org/matrix-react-sdk/pull/10790…

File: res/css/views/rooms/_TopUnreadMessagesBar.pcss [hunk 1]
```diff
@@ -22,60 +22,73 @@ limitations under the License.
...
-    cursor: pointer;
-}
+    /* Define common declarations. Their values are specified below. */
+    .mx_TopUnreadMessagesBar_scrollUp,
+    .mx_TopUnreadMessagesBar_markAsRead {
+        background: var(--background);
+        width: var(--width);
+        height: var(--height);
+        border: var(--border);
+        border-radius: var(--borderRadius);
```
Comment (luixxiul) [inline res/css/views/rooms/_TopUnreadMessagesBar.pcss:46]: Please note that using names which seem general for variables does not matter here, as the custom properties cascade. Even if `--height`, for example, is specified somewhere outside of this file, its value is not be respected, because the value for the custom property is explicitly specified below for each selector (`--height: 38px` and `--height: 18px`).

File: res/css/views/rooms/_TopUnreadMessagesBar.pcss [hunk 2]
```diff
@@ -22,60 +22,73 @@ limitations under the License.
...
+        --height: 18px;
+        --borderRadius: 10px;
+
+        display: block;
+        margin: 5px auto;
+
+        &::before {
+            --maskImage: url("$(res)/img/cancel.svg");
+            --maskSize: 10px;
+        }
```
Comment (luixxiul) [inline res/css/views/rooms/_TopUnreadMessagesBar.pcss:92]: `--width: 18px` and `--height: 18px` do not need to be repeated here, as they are inherited from `mx_TopUnreadMessagesBar_markAsRead`.

File: res/css/views/rooms/_TopUnreadMessagesBar.pcss [hunk 3]
```diff
@@ -22,60 +22,73 @@ limitations under the License.
...
-    height: 38px;
-    border-radius: 19px;
-    box-sizing: border-box;
-    background: $background;
-    border: 1.3px solid $muted-fg-color;
-    cursor: pointer;
-}
+    /* Define common declarations. Their values are specified below. */
+    .mx_TopUnreadMessagesBar_scrollUp,
+    .mx_TopUnreadMessagesBar_markAsRead {
```
Comment (richvdh) [inline res/css/views/rooms/_TopUnreadMessagesBar.pcss:41]: what is the reason for nesting these where they were not before?

Comment (luixxiul) [reply res/css/views/rooms/_TopUnreadMessagesBar.pcss:41]: To avoid repeating the same declarations, and to make the structure compact. Do you rather want to write like this?

````
    .mx_TopUnreadMessagesBar_scrollUp {
        background: var(--background);
        width: var(--width);
        height: var(--height);
        border: var(--border);
        border-radius: var(--borderRadius);

        --background: $background;
        --border: 1.3px solid $muted-fg-color;

        &::before {
            content: "";
            position: absolute;
            background: var(--background);
            width: var(--width);
            height: var(--height);
            mask-repeat: no-repeat;
            mask-position: center;
            mask-image: var(--maskImage);
            mask-size: var(--maskSize);

            --background: $muted-fg-color;
        }
    }

    .mx_TopUnreadMessagesBar_markAsRead {
        background: var(--background);
        width: var(--width);
        height: var(--height);
        border: var(--border);
        border-radius: var(--borderRadius);

        --background: $background;
        --border: 1.3px solid $muted-fg-color;

        &::before {
            content: "";
            position: absolute;
            background: var(--background);
            width: var(--width);
            height: var(--height);
            mask-repeat: no-repeat;
            mask-position: center;
            mask-image: var(--maskImage);
            mask-size: var(--maskSize);

            --background: $muted-fg-color;
        }
    }
````

It works, but it's not really efficient.

Comment (luixxiul) [reply res/css/views/rooms/_TopUnreadMessagesBar.pcss:41]: Also, there seems to me that there has not been a specific reason either that those selectors were not combined, checking a commit like https://github.com/matrix-org/matrix-react-sdk/pull/2345/commits/ac9902e52a704f52e508757fef7cf8c9b517f11b#diff-21b2361c0ea031da22f3680c5912c17ccc3d86c5bad901dd533f004cde3efa32 of https://github.com/matrix-org/matrix-react-sdk/pull/2345 and https://github.com/matrix-org/matrix-react-sdk/pull/4159/commits/07348a65ad66b733c6de899ade6ba47c71c0e710#diff-21b2361c0ea031da22f3680c5912c17ccc3d86c5bad901dd533f004cde3efa32 of https://github.com/matrix-org/matrix-react-sdk/pull/4159.

Comment (richvdh) [reply res/css/views/rooms/_TopUnreadMessagesBar.pcss:41]: I'm afraid I don't understand your answer. I think you are explaining why `.mx_TopUnreadMessagesBar_scrollUp` and `.mx_TopUnreadMessagesBar_markAsRead` are *combined*, but I didn't ask that. I asked why they are *nested* (inside `.mx_TopUnreadMessagesBar`).
